File: test/data/modifier_audit/async_functions.ts
async function fetchData(url: string): Promise<string> {
  return await fetch(url).then(r => r.text());
}

function syncHelper(): number {
  return 42;
}

export async function exportedAsync(): Promise<void> {
  await Promise.resolve();
}

class MyService {
  async start(): Promise<void> {}
  stop(): void {}
  static async create(): Promise<MyService> { return new MyService(); }
}

File: test/data/modifier_audit/async_functions.rs
async fn fetch_data(url: &str) -> String {
    reqwest::get(url).await.unwrap().text().await.unwrap()
}

fn sync_helper() -> i32 {
    42
}

pub async fn public_async(item: &str) -> bool {
    true
}

unsafe fn unsafe_helper() {
    std::ptr::null::<i32>();
}

pub fn public_sync() -> &'static str {
    "hello"
}
